docs/directives/needuml.rst

```diff
@@ -152,6 +152,39 @@ and can't be ``diagram``.
       B -> C: Hi
       C -> B: Hi there
 
+save
+~~~~
+
+Allows to store generated Plantuml-code of ``needuml`` to given file path. The given file path must be
+relative path to ``conf.py`` directory. For example, ``_build/_out/my_needuml.puml``.
```

Comment:
Hello @haiyangToAI,
Really cool Feature and implementation.
I have my doubts to store the puml relative to the conf.py. I believe ist shall be relative to the out folder. 
What do you think?
Nice regards

Reply:
For me it's hard to answer this question, as I see use cases and drawbacks for both locations.

My first thought was, that storage should be independent of the build location.
So  a ``make html`` stores the file at the same location as a ``make json``. Maybe somewhere on the root folder of the project, outside the docs folder.

But then I thought, what if it shall be part for the build artifact. Then this will never work.

Another question was: Do we really need to store the plantuml-output for each doc-build (and overwrite existing files)?

I came to the conclusion, that may be a special sphinx-builder would be better. For instance: ``make needumls``.
This then creates all puml files under ``_build/needumls``.

Maybe combined with a new config option `needs_build_needumls`, which takes a path, which is relative to the build-location. So it can be triggered  during a normal ``make html`` as well (similar to creating the needs.json during a HTML build.)

@PhilipPartsch: What do you think?

@haiyangToAI: Sorry for adding new ideas to this PR :/

Reply:
I really like the Idea of dedicated builder would be helpful and clean.

Reply:
I also like the idea. I will adapt the implementation. No worries.